subsys/net/lib/openthread/platform/logging.c

```diff
@@ -39,13 +39,33 @@ static inline int log_translate(otLogLevel aLogLevel)
 	return -1;
 }
 
+static uint32_t count_args(const char *fmt)
+{
+	uint32_t args = 0U;
+	bool prev = false; /* if previous char was a modificator. */
+
+	while (*fmt != '\0') {
+		if (*fmt == '%') {
+			prev = !prev;
+		} else if (prev) {
+			args++;
+			prev = false;
+		} else {
+			; /* standard character, continue walk */
+		}
+		fmt++;
+	}
+
+	return args;
+}
+
 void otPlatLog(otLogLevel aLogLevel, otLogRegion aLogRegion, const char *aFormat, ...)
 {
 	ARG_UNUSED(aLogRegion);
 
 #if defined(CONFIG_LOG)
 	int level = log_translate(aLogLevel);
-	uint32_t args_num = log_count_args(aFormat);
+	uint32_t args_num = count_args(aFormat);
```

Comment:
Is there compiler warning about unused function if CONFIG_LOG is not set?

Reply:
It does, thanks for catching. Fixed now.